From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Mon, 4 Jan 2021 17:15:14 +0100
Subject: Fix 32-bit test case

Force difference of microseconds to be computed in long long

Panic: file time-util.c: line 65 (timeval_cmp_margin): assertion failed: (usecs_diff >= 0)
---
 src/lib/time-util.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/lib/time-util.c b/src/lib/time-util.c
index 294bb02..ded1e93 100644
--- a/src/lib/time-util.c
+++ b/src/lib/time-util.c
@@ -43,17 +43,17 @@ int timeval_cmp_margin(const struct timeval *tv1, const struct timeval *tv2,
 
 	if (tv1->tv_sec < tv2->tv_sec) {
 		sec_margin = ((int)usec_margin / 1000000) + 1;
-		if ((tv2->tv_sec - tv1->tv_sec) > sec_margin)
+		if (((long long)tv2->tv_sec - tv1->tv_sec) > sec_margin)
 			return -1;
-		usecs_diff = (tv2->tv_sec - tv1->tv_sec) * 1000000LL +
-			(tv2->tv_usec - tv1->tv_usec);
+		usecs_diff = ((long long)tv2->tv_sec - tv1->tv_sec) * 1000000LL +
+			((long long)tv2->tv_usec - tv1->tv_usec);
 		ret = -1;
 	} else if (tv1->tv_sec > tv2->tv_sec) {
 		sec_margin = ((int)usec_margin / 1000000) + 1;
-		if ((tv1->tv_sec - tv2->tv_sec) > sec_margin)
+		if (((long long)tv1->tv_sec - tv2->tv_sec) > sec_margin)
 			return 1;
-		usecs_diff = (tv1->tv_sec - tv2->tv_sec) * 1000000LL +
-			(tv1->tv_usec - tv2->tv_usec);
+		usecs_diff = ((long long)tv1->tv_sec - tv2->tv_sec) * 1000000LL +
+			((long long)tv1->tv_usec - tv2->tv_usec);
 		ret = 1;
 	} else if (tv1->tv_usec < tv2->tv_usec) {
 		usecs_diff = tv2->tv_usec - tv1->tv_usec;
